hoomd/pytest/test_box_variant.py

```diff
@@ -0,0 +1,172 @@
+        test_box.volume = (1 - _f) * _test_box1.volume + _f * final_volume
+        npt.assert_allclose(box_t(custom_variant, _t), box_to_array(test_box))
+
+
+def test_interpolate_evaluation():
+    t_start = 50
+    t_ramp = 100
+    scalar_variant = hoomd.variant.Ramp(0, 1, t_start, t_ramp)
+    box_variant = hoomd.variant.box.Interpolate(_test_box1, _test_box2,
+                                                scalar_variant)
+    npt.assert_allclose(box_variant(0), box_to_array(_test_box1))
+    npt.assert_allclose(box_variant(25), box_to_array(_test_box1))
+    npt.assert_allclose(box_variant(t_start), box_to_array(_test_box1))
+
+    npt.assert_allclose(
+        box_variant(51),
+        0.99 * box_to_array(_test_box1) + 0.01 * box_to_array(_test_box2))
+    npt.assert_allclose(
+        box_variant(75),
+        0.75 * box_to_array(_test_box1) + 0.25 * box_to_array(_test_box2))
+    npt.assert_allclose(
+        box_variant(100),
+        0.5 * box_to_array(_test_box1) + 0.5 * box_to_array(_test_box2))
+    npt.assert_allclose(
+        box_variant(125),
+        0.25 * box_to_array(_test_box1) + 0.75 * box_to_array(_test_box2))
+    npt.assert_allclose(
+        box_variant(149),
+        0.01 * box_to_array(_test_box1) + 0.99 * box_to_array(_test_box2))
+
+    npt.assert_allclose(box_variant(t_start + t_ramp), box_to_array(_test_box2))
+    npt.assert_allclose(box_variant(t_start + t_ramp + 100),
+                        box_to_array(_test_box2))
+    npt.assert_allclose(box_variant(t_start + t_ramp + 1000000),
+                        box_to_array(_test_box2))
+
+
+def test_inverse_volume_ramp_evaluation():
+    box1 = hoomd.Box(10, 10, 10, 0.1, 0.2, 0.3)
+    final_volume = 500
+    t_start = 10
+    t_ramp = 100
+    variant = hoomd.variant.box.InverseVolumeRamp(box1, final_volume, t_start,
+                                                  t_ramp)
+
+    def get_volume(variant, timestep):
+        return hoomd.Box(*variant(timestep)).volume
+
+    assert get_volume(variant, 0) == box1.volume
+    assert get_volume(variant, 5) == box1.volume
+    assert get_volume(variant, 10) == box1.volume
+    assert get_volume(variant, 11) != box1.volume
+    npt.assert_allclose(get_volume(variant, 35),
+                        (0.75 / box1.volume + 0.25 / final_volume)**-1)
+    npt.assert_allclose(get_volume(variant, 60),
+                        (0.5 / box1.volume + 0.5 / final_volume)**-1)
+    npt.assert_allclose(get_volume(variant, 85),
+                        (0.25 / box1.volume + 0.75 / final_volume)**-1)
+    npt.assert_allclose(get_volume(variant, 110), final_volume)
+    npt.assert_allclose(get_volume(variant, 1010), final_volume)
+    # make sure tilts don't change
+    npt.assert_allclose(box1.tilts, variant(0)[3:])
+    npt.assert_allclose(box1.tilts, variant(5)[3:])
+    npt.assert_allclose(box1.tilts, variant(25)[3:])
+    npt.assert_allclose(box1.tilts, variant(125)[3:])
+    npt.assert_allclose(box1.tilts, variant(625)[3:])
```

Comment:
```suggestion
    for step in (0, 5, 252 125, 625):
        npt.assert_allclose(box1.tilts, variant(step)[3:])
```
There are a lot of opportunities to shorten the code with for loops.